geps/gep-91/index.md

```diff
@@ -1,22 +1,122 @@
+| Kong           | [mTLS Plugin](https://docs.konghq.com/hub/kong-inc/mtls-auth/)      |
+| Traefik        | [TLSOption.Spec.ClientAuth](https://doc.traefik.io/traefik/https/tls/#client-authentication-mtls)    |
+
+### API
+
+* Introduce a `clientValidation` field of type `ClientValidationContext` within [GatewayTLSConfig][] that can be used to validate the client initiating the TLS connection
+to the Gateway.
+* Introduce a `caCertificateRefs` field within `ClientValidationContext` that can be used to specify a list of CA Certificates that
+can be used as a trust anchor to validate the certificates presented by the client.
+* This new field is mutually exclusive with the [BackendTLSPolicy][] configuation which is used to validate the TLS certificate presented by the peer on the connection between the Gateway and the backend, and this GEP is adding support for validating the TLS certificate presented by the peer on the connection between the Gateway and the downstream client.
+
+#### GO
+
+```go
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
+    // CACertificateRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates of
+    // the Certificate Authorities that can be used
+    // as a trust anchor to validate the certificates presented by the client.
+    //
+    // A single CA certificate reference to a Kubernetes ConfigMap kind has "Core" support.
+    // Implementations MAY choose to support attaching multiple CA certificates to
+    // a Listener, but this behavior is implementation-specific.
+    //
+    // Support: Core - An optional single reference to a Kubernetes Secret and ConfigMap,
+    // with the CA certificate in a key named `ca.crt`.
+    //
+    // Support: Implementation-specific (More than one reference, or other kinds
+    // of resources).
+    //
+    // References to a resource in a different namespace are invalid UNLESS there
+    // is a ReferenceGrant in the target namespace that allows the certificate
+    // to be attached. If a ReferenceGrant does not allow this reference, the
+    // "ResolvedRefs" condition MUST be set to False for this listener with the
+    // "RefNotPermitted" reason.
+    //
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:validation:MinItems=1
+    CACertificateRefs []SecretObjectReference `json:”caCertificateRefs,omitempty”`
+    // SubjectAltNames contains one or more alternate names to verify
+    // the subject identity in the certificate presented by the client.
+    //
+    // Support: Core
+    //
+    // +optional
+    SubjectAltNames []string `json:"subjectAltNames,omitempty"`
```

Comment:
@arkodg See https://github.com/kubernetes-sigs/gateway-api/pull/2759#discussion_r1483295712
```suggestion
    // +kubebuilder:validation:MinItems=1
    SubjectAltNames []string `json:"subjectAltNames,omitempty"`
```

Reply:
afaik no implementation make SAN mandatory (only CA Cert is mandatory), curious why we want to do it here ?

Reply:
Maybe I'm mistaken, but IIUC we have `// +optional` which says the field is not required, but `MinItems=1` means that if the field is set then there must be at least 1 entry.

Reply:
@arkodg I tried it (`+optional` and `MinItems=1`) and it works as I said above, i.e., SAN is not mandatory.

Reply:
awesome thanks for checking, will update